subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/internal/LanguageSpecificAdaptor.java

```diff
@@ -151,7 +152,7 @@ private BuildScriptBuilder projectBuildScriptBuilder(String projectName, InitSet
 
     private BuildScriptBuilder conventionPluginScriptBuilder(String conventionPluginName, InitSettings settings) {
         BuildScriptBuilder buildScriptBuilder = scriptBuilderFactory.script(settings.getDsl(),
-            "buildSrc/src/main/" + settings.getDsl().name().toLowerCase() + "/" + settings.getPackageName() + "." + getLanguage().getName() + "-" + conventionPluginName + "-conventions");
+            "buildSrc/src/main/" + TextUtil.toLowerCaseUserLocale(settings.getDsl().name()) + "/" + settings.getPackageName() + "." + getLanguage().getName() + "-" + conventionPluginName + "-conventions");
```

Comment:
I believe this needs to be toLowerCaseLocaleSafe because in Turkish locales, we would end up creating a `buildSrc/src/main/kotlın` (dotless i) directory.